src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -23,6 +24,11 @@ static string GetRoutingKey(Type type, string key = "")
 
             var implementedInterface = interfaces.FirstOrDefault();
 
+            if (interfaces.Count > 0)
+            {
+                Logger.WarnFormat("The direct routing topology might not deliver message types with interfaces in their inheritance hierarchy. Refrain from using interfaces for routing purposes. Interfaces found are {0}, the one that will be used for building routing key is {1}", string.Join(", ", interfaces), implementedInterface);
+            }
+            
```

Comment:
Also, a change to say something like "Message types used for publishing should not implement interfaces" instead of the "refrain" sentence.